File: B_Rudolph_and_Tic_Tac_Toe.py
t=int(input())
for _ in range(t):
    game=[]
    m={}
    for _ in range(3):
        row=input().strip()
        game.append(row)
        m[row]=1


    for i in range(3):
        col=game[0][i]+game[1][i]+game[2][i]
        m[col]=1

    diag="".join(game[i][i] for i in range(3))

    m[diag]=1

    antidiag="".join(game[i][3-1-i] for i in range(3))
    
    m[antidiag]=1

    ans="DRAW"

    if("XXX" in m): ans="X"
    if("OOO" in m): ans="O"
    if("+++" in m): ans="+"

    print(ans)

File: B_Rudolph_and_Tic_Tac_Toe.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    int t;
    cin>>t;
    while(t--){
        vector<string> arr;
        unordered_map<string,int> m;
        for(int i=0; i<3; i++){
            string row;
            cin>>row;
            arr.push_back(row);
            m[row]=1;
        }

        string col="";

        for(int i=0; i<3; i++){
            col+=arr[0][i];
            col+=arr[1][i];
            col+=arr[2][i];
            m[col]=1;
            col="";
        }

        string diag="";

        for(int i=0; i<3; i++){
            diag+=arr[i][i];
        }

        m[diag]=1;

        string antidiag="";

        for(int i=0; i<3; i++){
            antidiag+=arr[i][3-1-i];
        }

        m[antidiag]=1;

        string ans="DRAW";

        if(m["XXX"]) ans="X";
        if(m["OOO"]) ans="O";
        if(m["+++"]) ans="+";

        cout<<ans<<endl;
    }
}